src/server/ua_services_method.c

```diff
@@ -178,6 +179,95 @@ callWithMethodAndObject(UA_Server *server, UA_Session *session,
             }
         }
     }
+    if(!found) {
+        /* If the object doesn't have a hasComponent reference to the method node,
+         * check its objectType (and its supertypes). Invoked method can be a component
+         * of objectType and be invoked on this objectType's instance (or on a instance
+         * of one of its subtypes). */
+
+        UA_ReferenceTypeSet hasTypeDefinitionRefs;
```

Comment:
You can use the existing
```c
const UA_Node *
getNodeType(UA_Server *server, const UA_NodeHead *head);
```

To get the node type. This will be the ObjectType if it is an object node.

Reply:
Thanks, somehow I missed this function